writer.go

```diff
@@ -704,10 +730,42 @@ func (w *writer) write(conn *Conn, batch []Message, resch [](chan<- error)) (ret
 
 	t0 := time.Now()
 	conn.SetWriteDeadline(time.Now().Add(w.writeTimeout))
-	if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
-		w.stats.errors.observe(1)
+	attempts := 0
+	for {
+		if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
```

Comment:
if `err == nil` then you're not breaking out of the for loop

Reply:
yep